app/scripts/background.js

```diff
@@ -266,13 +266,17 @@ function maybeDetectPhishing(theController) {
       }
 
       theController.phishingController.maybeUpdateState();
-      const phishingTestResponse = theController.phishingController.test(
-        details.url,
-      );
 
       const blockedRequestResponse =
         theController.phishingController.isBlockedRequest(details.url);
 
+      let phishingTestResponse;
+      if (details.type === 'main_frame' || details.type === 'sub_frame') {
+        phishingTestResponse = theController.phishingController.test(
```

Comment:
I think you have this backwards??

`isBlockedRequest` is for C2s, but `test` is for the EPD list.

Reply:
Right, but this allowed us to add C2s to EPD which would enable SEAL, however we want SEAL to have to directly send us the C2s, so we dont want to check the EPD list against network requests which is what is happening.

Reply:
C2 requests can still come from `main_frame` and `sub_frame` though, so how does this solve that?

Reply:
network requests only come from `xmlhttprequest` which we do not want EPD to check against, however we want EPD to check again `main_frame` and `sub_frame`

Reply:
It's okay if EPD blocks a C2 from a main_frame or sub_frame.